src/message/connection.js

```diff
@@ -531,7 +531,8 @@ Connection.prototype._tryReconnect = function () {
   } else {
     this._clearReconnect()
     this.close()
-    this._client.emit(C.MAX_RECONNECTION_ATTEMPTS_REACHED, this._reconnectionAttempt)
+    this._setState(C.CONNECTION_STATE.CLOSED)
```

Comment:
Was there an issue that motivated the call to `_setState`? 
The state is also updated asynchronously in the call to `_onClose` (once the endpoint closes).